ghost/link-tracking/lib/LinkClickTrackingService.js

```diff
@@ -78,7 +97,102 @@ class LinkClickTrackingService {
         });
     }
 
-    /** 
+    /**
+   * validate and manage the new redirect url in filter
+   * `to` url needs decoding and transformation to relative url for comparision
+   * @param {string} filter
+   * @returns {Object} parsed filter
+   * @throws {errors.BadRequestError}
+   */
+    #parseLinkFilter(filter) {
+        // decode filter to manage any encoded uri components
+        filter = decodeURIComponent(filter);
+
+        try {
+            const filterJson = nql(filter).parse();
+            const postId = filterJson?.$and?.[0]?.post_id;
+            const redirectUrl = new URL(filterJson?.$and?.[1]?.to);
+            if (!postId || !redirectUrl) {
```

Comment:
Should we be using an NQL filter if we're going to be restricting it to such a particular format? Would it be simpler to pass a `postId` and and a `redirectUrl` in the body/query? 🤔

Reply:
The reason i'd stick to keeping the NQL filter is because that uses our standard bulk endpoint  pattern, while passing stuff in body will make it really custom. We currently we have limited examples - links and members - for this pattern, but i am planning to update the docs with it this week.

Reply:
Do we have similar restrictions on the filters for the members bulk endpoints?

Reply:
Don't think so, but the usecase here is different as in case of Members the bulk edits do add/remove label based on the filter which can be anything and isn't dependent on label being added/removed. While here if the bulk action is `updateLink` its necessary to have the old link and post id in place as you shouldn't be able to bulk update without it.